modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/model/impl/CPDefinitionDiagramSettingImpl.java

```diff
@@ -14,24 +14,48 @@
 
 package com.liferay.commerce.shop.by.diagram.model.impl;
 
+import com.liferay.commerce.product.model.CPAttachmentFileEntry;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.commerce.product.service.CPDefinitionLocalServiceUtil;
+import com.liferay.commerce.shop.by.diagram.constants.CPDefinitionDiagramSettingsConstants;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.List;
+
 /**
- * The extended model implementation for the CPDefinitionDiagramSetting service. Represents a row in the &quot;CPDefinitionDiagramSetting&quot; database table, with each column mapped to a property of this class.
- *
- * <p>
- * Helper methods and all application logic should be put in this class. Whenever methods are added, rerun ServiceBuilder to copy their definitions into the <code>com.liferay.commerce.shop.by.diagram.model.CPDefinitionDiagramSetting</code> interface.
- * </p>
- *
  * @author Andrea Sbarra
+ * @author Alessio Antonio Rendina
  */
 public class CPDefinitionDiagramSettingImpl
 	extends CPDefinitionDiagramSettingBaseImpl {
 
-	/**
-	 * NOTE FOR DEVELOPERS:
-	 *
-	 * Never reference this class directly. All methods that expect a cp definition diagram setting model instance should use the {@link com.liferay.commerce.shop.by.diagram.model.CPDefinitionDiagramSetting} interface instead.
-	 */
 	public CPDefinitionDiagramSettingImpl() {
 	}
 
+	@Override
+	public CPAttachmentFileEntry fetchCPAttachmentFileEntry()
+		throws PortalException {
+
+		CPDefinition cpDefinition =
+			CPDefinitionLocalServiceUtil.getCPDefinition(getCPDefinitionId());
```

Comment:
@Commando18 fetch methods should never throw PortalException. You should change fetchCPAttachmentFileEntry to fetch cpDefinition and return null if no cpDefinition.
fetch should return object if it exists or null in any other case. If it is unusual that needed cpDefinition misses, you can log.warning it.

Reply:
You're right. I will change fetch CPDefinition then and add a log. Thanks

Reply:
I cannot do this because the getter of the attachments list is throwing PortalException, so I changed it to getCPAttachmentFileEntry and throw NoSuchCPAttachmentFileEntryException if list is empty. Let me know if you're good with this solution.
Thanks!